### DISCUSSION TITLE
Cosplay动漫人物生成时报错

### DISCUSSION THREAD

**Discussion by RuipingWang1986** [modelscope/ms-agent#478] [Q&A]
"
import os
import sys
import time
import requests
from pathlib import PurePosixPath

# 配置环境变量；如果您已经提前将api-key提前配置到您的运行环境中，可以省略这个步骤
os.environ['DASHSCOPE_API_KEY'] = 'our_DASHSCOPE_API_KEY'  # 替换为您的 Dashscope API Key
dashscope_api_key = os.getenv('DASHSCOPE_API_KEY')
os.environ['SM_MS_API_KEY'] = 'our_SM_MS_API_KEY'  # 替换为您的 sm.ms API Key
sm_ms_api_key = os.getenv('SM_MS_API_KEY')

# 用于存储对话历史记录的列表
dialog_history = []

def upload_image(file_path):
    url = 'https://sm.ms/api/v2/upload'
    headers = {'Authorization': sm_ms_api_key}
    files = {'smfile': open(file_path, 'rb')}

    response = requests.post(url, headers=headers, files=files)
    if response.status_code == 200:
        response_json = response.json()
        if response_json['success']:
            print(f"上传成功，图像 URL: {response_json['data']['url']}")
            return response_json['data']['url']
        elif response_json['code'] == 'image_repeated':
            print(f"图像已经存在，图像 URL: {response_json['images']}")
            return response_json['images']
        else:
            raise Exception(f"上传失败：{response_json['message']}")
    else:
        response.raise_for_status()

def generate_image(face_image_url, template_image_url):
    # 请求参数准备数据
    data = {
        "model": "wanx-style-cosplay-v1",
        "input": {
            "model_index": 1,
            "face_image_url": face_image_url,
            "template_image_url": template_image_url
        }
    }

    print(f"请求参数: {data}")

    # 发起请求
    rsp = requests.post(
        url='https://dashscope.aliyuncs.com/api/v1/services/aigc/image-generation/generation',
        headers={
            'X-DashScope-Async': 'enable',
            'Authorization': f'Bearer {dashscope_api_key}',
            'Content-Type': 'application/json'
        },
        json=data
    )

    # 调试信息：打印完整响应
    print(f'HTTP 状态码: {rsp.status_code}')
    print(f'响应内容: {rsp.text}')

    if rsp.status_code == 200:
        response_json = rsp.json()
        task_id = response_json['output']['task_id']
        return task_id
    else:
        raise Exception(
            f'生成失败，状态码: {rsp.status_code}, 错误代码: {rsp.json().get("code")}, 错误信息: {rsp.json().get("message")}')

def get_generated_image(task_id):
    url = 'https://dashscope.aliyuncs.com/api/v1/services/aigc/image-generation/generation/results'
    headers = {
        'Authorization': f'Bearer {dashscope_api_key}',
        'Content-Type': 'application/json'
    }

    data = {
        "task_id": task_id,
        "model": "wanx-style-cosplay-v1"
    }

    while True:
        rsp = requests.post(url, headers=headers, json=data)

        if rsp.status_code == 200:
            response_json = rsp.json()
            task_status = response_json["output"]["task_status"]
            if task_status == 'FINISHED':
                return response_json['output']['generated_images'][0]['url']
            elif task_status == 'FAILED':
                raise Exception(f'生成任务失败：{response_json["output"]["message"]}')
            else:
                print(f'任务状态为 {task_status}，等待10秒钟后重试...')
                time.sleep(10)  # 等待一段时间再重试
        else:
            response_json = rsp.json()
            raise Exception(
                f'获取任务状态失败，状态码: {rsp.status_code}, 错误代码: {response_json.get("code")}, 错误信息: {response_json.get("message")}')

def main():
    print("您好，我是图像合成助手，能够帮助您生成图片。请告诉我您想要生成的图片内容：")

    while True:
        try:
            # 获取用户输入
            face_image_path = input('请输入本地人脸图像路径：')
            template_image_path = input('请输入本地模板图像路径：')
            if face_image_path.lower() == '退出' or template_image_path.lower() == '退出':
                print("程序已退出。")
                break

            # 上传图像并获取URL
            face_image_url = upload_image(face_image_path)
            template_image_url = upload_image(template_image_path)

            # 检查上传路径
            print(f"已上传并确认的图像路径 - Face: {face_image_url}, Template: {template_image_url}")

            # 使用函数处理用户输入的问题
            task_id = generate_image(face_image_url, template_image_url)
            print(f"任务已提交，任务ID: {task_id}")

            # 获取生成图像状态
            response = get_generated_image(task_id)
            print(f"生成的图像 URL: {response}")

        except KeyboardInterrupt:
            # 如果用户按下Ctrl+C，则退出程序
            print("\n程序已退出。")
            break
        except Exception as e:
            # 打印错误信息并继续
            print(f"发生错误：{e}", file=sys.stderr)
            continue

if __name__ == "__main__":
    main()"
运行上述代码后，出现如下错误：
您好，我是图像合成助手，能够帮助您生成图片。请告诉我您想要生成的图片内容：
请输入本地人脸图像路径：wrp.jpg
请输入本地模板图像路径：SDS.jpg
图像已经存在，图像 URL: https://s2.loli.net/2024/06/11/XokMJrFq1ilu8fW.jpg
图像已经存在，图像 URL: https://s2.loli.net/2024/06/11/QqfxCBmgtdo7XWF.jpg
已上传并确认的图像路径 - Face: https://s2.loli.net/2024/06/11/XokMJrFq1ilu8fW.jpg, Template: https://s2.loli.net/2024/06/11/QqfxCBmgtdo7XWF.jpg
请求参数: {'model': 'wanx-style-cosplay-v1', 'input': {'model_index': 1, 'face_image_url': 'https://s2.loli.net/2024/06/11/XokMJrFq1ilu8fW.jpg', 'template_image_url': 'https://s2.loli.net/2024/06/11/QqfxCBmgtdo7XWF.jpg'}}
HTTP 状态码: 200
响应内容: {"output":{"task_status":"PENDING","task_id":"3b7305ae-8fce-4500-8968-3a8c1f47a05c"},"request_id":"9566d7d4-1489-96ec-994e-645372cc2810"}
任务已提交，任务ID: 3b7305ae-8fce-4500-8968-3a8c1f47a05c
**发生错误：获取任务状态失败，状态码: 400, 错误代码: InvalidParameter, 错误信息: url error, please check url！**能否麻烦帮忙看一下如何解决，或者有没有可供运行的python代码呢？

**Comment by mushenL** [comment]
您好，按照您提供的request_id和task_id看，图像生成任务是成功的，应该是查询任务的dashone URL不太对
我这边看查询的url应该是如下这个
![image](https://github.com/modelscope/modelscope-agent/assets/125954878/beba6057-3f30-413b-84f0-a68bde8f3860)
